common.lic

```diff
@@ -471,6 +471,18 @@ module DRC
     end
   end
 
+  def check_listening
+    return if DRRoom.pcs.empty?
+    classes = assess_teach
+    @last_teacher = classes
+                    .reject { |t, s| t.nil? || s.nil? }
+                    .sort_by { |_t, s| [DRSkill.getxp(s), DRSkill.getrank(s)] }
+                    .find { |t, _s| listen?(t) }
+                    .first
+
+    @listen_timer = @last_teacher ? nil : Time.now
```

Comment:
Same comment as above about instance variables. This line should be removed. If tracking is helpful then the calling script probably should handle it is my preference to help keep the common methods stateless.